drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -220,6 +222,16 @@ int i2c_stm32_target_register(const struct device *dev,
 		return ret;
 	}
 
+#if defined(CONFIG_PM_DEVICE_RUNTIME)
+	if (pm_device_wakeup_is_capable(dev)) {
+		/* Mark device as active */
+		(void)pm_device_runtime_get(dev);
+		/* Enable wake-up from stop */
+		LOG_DBG("i2c: enabling wakeup from stop");
+		LL_I2C_EnableWakeUpFromStop(cfg->i2c);
```

Comment:
Why having a separate treatment from non target mode?
Supporting on non target mode is another possible use case which can be implemented. Having it supported this way on target mode, makes default mode implementation trickier (or at least can lead to conflict)

Reply:
Wake up from STOP is only supported (and relevant) in target mode. It enables the functionality to wake up the MCU when a read from or write to the configured target address is detected on the bus.

Reply:
Makes perfect sense